Sprint 4

elizavetamakarovavn-netizen opened the pull request.

elizavetamakarovavn-netizen committed:
pom

elizavetamakarovavn-netizen committed:
sprint 4

Ayantea1 commented on src/main/java/org/example/Main.java:
```diff
@@ -0,0 +1,17 @@
+package org.example;
```
Нужно исправить: в проекте не должно быть лишнего кода

Ayantea1 commented on src/main/java/ru/yandex/praktikum/scooter/pageobject/MainPage.java:
```diff
@@ -0,0 +1,81 @@
+
+    // Клик по кнопке "Заказать" сверху
+    public void clickTopOrderButton() {
+        WebElement button = driver.findElement(By.xpath("//button[text()='Заказать']"));
```
Нужно исправить: все локаторы должны быть вынесены

Ayantea1 commented on src/test/java/ru/yandex/praktikum/scooter/tests/FaqTest.java:
```diff
@@ -0,0 +1,57 @@
+        WebDriver driver = factory.getDriver();
+        MainPage mainPage = factory.getMainPage();
+        String actualAnswer = mainPage.getFaqAnswerByIndex(questionIndex);
+        assertEquals("Ответ на вопрос №" + questionIndex + " некорректный",
```
можно улучшить: первым у тебя будет вопрос 0, не очень хороший вариант для нумерации

Ayantea1 commented on src/test/java/ru/yandex/praktikum/scooter/tests/OrderButtonTest.java:
```diff
@@ -0,0 +1,49 @@
+
+        WebElement header = new WebDriverWait(driver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.visibilityOfElementLocated(
+                        By.xpath("//div[text()='Для кого самокат']")));
```
Нужно исправить: в тестах локаторов быть не должно

Ayantea1 commented on src/main/java/ru/yandex/praktikum/scooter/pageobject/OrderPage.java:
```diff
@@ -0,0 +1,154 @@
+
+    // Проверка появления попапа "Заказ оформлен"
+    public String getOrderConfirmedHeaderText() {
+        return driver.findElement(orderConfirmedHeader).getText();
```
Можно улучшить: в самом локаторе у тебя указан текст. Методом ты получаешь тот текст, по которому искал, и его же ожидаешь получить в тесте. Выглядит как лишние элементы в цепочке, достаточно проверить, что находится элемент с текстом

Ayantea1 commented on src/main/java/ru/yandex/praktikum/scooter/pageobject/MainPage.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    // Верхняя кнопка "Заказать"
+    private final By topOrderButton = By.xpath("//button[@class='Button_Button__ra12g']");
```
Нужно исправить: не должно быть неиспользуемого кода

elizavetamakarovavn-netizen committed:
1. Удален класс Main.
2. В методе clickTopOrderButton() (MainPage.java) убран локатор (+ не осталось неиспользуемого кода).
3. В FaqTest.java откорректирована формулировка ошибки.
4. В OrderButtonTest.java удалены локаторы.
5. В OrderPage.java откорректированы методы проверки появления попапов.

Ayantea1 commented on src/test/java/ru/yandex/praktikum/scooter/tests/OrderFormTest.java:
```diff
@@ -77,15 +77,14 @@ public void testOrderFlow() {
-                modalText.contains("Хотите оформить заказ"));
-
+        boolean isModalDisplayed = orderPage.isModalHeaderDisplayed();
+        assertTrue("Ожидалось, что появится попап 'Хотите оформить заказ', но он не появился", isModalDisplayed);
```
Желательно проверять ассертом только то, для чего написан тест

Ayantea1 commented on src/test/java/ru/yandex/praktikum/scooter/tests/FaqTest.java:
```diff
@@ -1,12 +1,10 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
-import org.openqa.selenium.WebDriver;
+//import org.openqa.selenium.WebDriver;
```
Не оставляй закомментированный код

elizavetamakarovavn-netizen committed:
Удалён лишний файл Main.java

elizavetamakarovavn-netizen committed:
Откорректировано:

1. Удален закомментированный код в FaqTest.java.
2. Откорректирован метод 'Ожидание появления заголовка попапа "Хотите оформить заказ?"' в OrderPage.java.
3. Соответственно откорректирован тест OrderFormTest.java, используется только один assertTrue.

elizavetamakarovavn-netizen committed:
Revert "Удалён лишний файл Main.java"

This reverts commit 18186d234c3f6fc4177bbe814346717087a7bc0f.

elizavetamakarovavn-netizen committed:
Удалён лишний файл Main.java
